src/vcpkg/archives.cpp

```diff
@@ -181,4 +182,64 @@ namespace vcpkg
         Path to_path_partial = extract_archive_to_temp_subdirectory(paths, archive, to_path);
         fs.rename_with_retry(to_path_partial, to_path, VCPKG_LINE_INFO);
     }
+
+    int compress_directory_to_zip(const VcpkgPaths& paths, const Path& source, const Path& destination)
+    {
+        auto& fs = paths.get_filesystem();
+        fs.remove(destination, VCPKG_LINE_INFO);
+#if defined(_WIN32)
+        auto&& seven_zip_exe = paths.get_tool_exe(Tools::SEVEN_ZIP);
+
+        return cmd_execute_and_capture_output(
+                   Command{seven_zip_exe}.string_arg("a").path_arg(destination).path_arg(source / "*"),
+                   get_clean_environment())
+            .exit_code;
+
+#else
+        return cmd_execute_clean(Command{"zip"}
+                                     .string_arg("--quiet")
+                                     .string_arg("-y")
+                                     .string_arg("-r")
+                                     .path_arg(destination)
+                                     .string_arg("*")
+                                     .string_arg("--exclude")
+                                     .string_arg("\\*.DS_Store"),
```

Comment:
I do not like that we use `system` :|

Also, is there a reason this isn't just `.DS_Store`? My understanding is that `.DS_Store` will always be called exactly that.

Reply:
I thought it was needed to match the file in sub dirs, but that was not needed

Reply:
I think we should escape `*` in `string_arg()` (on platforms where the shell will interpret it). If the intent was to actually pass `*`, it should be a `raw_arg()`.